controllers/machinehealthcheck_controller.go

```diff
@@ -201,11 +201,13 @@ func (r *MachineHealthCheckReconciler) Reconcile(ctx context.Context, req ctrl.R
 	healthyCount := 0
 	for _, healthyTarget := range healthy {
 		log.Info("handling healthy target", "target", healthyTarget.String())
-		_, err := resourceManager.HandleHealthyNode(healthyTarget.Node.GetName(), healthyTarget.Machine.GetName(), mhc)
-		if err != nil {
+		if remainingCRs, err := resourceManager.HandleHealthyNode(healthyTarget.Node.GetName(), healthyTarget.Machine.GetName(), mhc); err != nil {
 			log.Error(err, "failed to handle healthy target", "target", healthyTarget.String())
 			return result, err
+		} else if len(remainingCRs) > 0 {
+			result.Requeue = true
 		}
+		// TODO consider to align with NHC, where nodes count as healtht only when all CRs are actuallt gone (e.g. remediator removed finalizer)
```

Comment:
Is there are a reason not to address this isue now in NHC?

NIT: Some typos in the comment (healtht, actuallt).

Reply:
> Is there are a reason not to address this isue now in NHC?

we want to have parity with MHC code for now, not NHC, so it would be a new feature and not a bug fix

Reply:
typos fixed, thx